src/arrayMethodJoin.js

```diff
@@ -5,7 +5,29 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let string = '';
+    let copySeparator = `${separator}`;
+
+    if (typeof separator === 'undefined') {
+      copySeparator = ',';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      switch (typeof this[i]) {
+        case 'object':
+          string += copySeparator;
+          break;
+        case 'undefined':
+          string += copySeparator;
```

Comment:
why do you need these two cases if the are the same?
And I am not sure you will have objesct as elements of array